lighthouse-core/audits/byte-efficiency/uses-long-cache-ttl.js

```diff
@@ -169,6 +160,36 @@ class CacheHeaders extends Audit {
     );
   }
 
+  /**
+   * @param {Map<string, string>} headers
+   * @param {ReturnType<import('parse-cache-control')>} cacheControl
+   * @param {?number} cacheLifetimeInSeconds
+   * @param {number} cacheHitProbability
+   * @returns {boolean}
+   */
+  static shouldProcessRecord(headers, cacheControl, cacheLifetimeInSeconds, cacheHitProbability) {
+    // The HTTP/1.0 Pragma header can disable caching if cache-control is not set, see https://tools.ietf.org/html/rfc7234#section-5.4
+    if ((headers.get('pragma') || '').includes('no-cache')) {
+      return false;
+    }
+
+    // Ignore assets where policy implies they should not be cached long periods
+    if (cacheControl &&
+      (
+        cacheControl['must-validate'] ||
+        cacheControl['no-cache'] ||
+        cacheControl['no-store'] ||
+        cacheControl['private'])) {
+      return false;
+    }
+
+    if (cacheLifetimeInSeconds !== null && cacheLifetimeInSeconds <= 0) return false;
+
+    if (cacheHitProbability > IGNORE_THRESHOLD_IN_PERCENT) return false;
```

Comment:
I'm wondering if we should just move the `cacheHitProbability` calculation to after this `shouldProcessRecord` and switch the ignore threshold to the equivalent time.

That might make it easier to figure out what's going on....

Reply:
> switch the ignore threshold to the equivalent time

I don't understand what you mean by this.

Reply:
cacheLifetimeInSeconds gets converted into a cacheHitProbability

instead of ignoring records with a `cacheHitProbability` above 92.5% what if we ignored records with `cacheLifetimeInSeconds` above 2 weeks (should be the equivalent of 92.5%)

Reply:
> I could be making an opposite suggestion to what @patrickhulce has suggested before, but to me there are two different sets of checks in here:

I didn't do a great job of clearly saying what I was lookin for before (sorry @Hoten !!) so it certainly could have been interpreted as opposite to that :)

My primary concern is not sprinkling around the different cases where things get ignored for the same category of reason, before this we checked `private` somewhere different from `no-cache` which are both ignored because of the signal in the `cache-control` directive.

I think it's relatively difficult to segment `shouldProcessRecord` and `computeCacheLifetime` as cleanly as you've done in that example @brendankenny  because there are some interdependencies, but if everyone's happy with some duplicated logic and think it's easy to read, I'm onboard.

IMO the fact that we're dealing with some mixture of `pragma`, `expires`, and `cache-control` all at the same time also complicates this, but I guess those two other cases are pretty simple.

Reply:
understood. I ran a bit too much with the idea of only using one `continue` ;)